Remotion/Data/DomainObjects/Mapping/Builder/ClassDefinitionBuilderNode.cs

```diff
@@ -105,5 +139,19 @@ private void AddDerivedClass (ClassDefinitionBuilderNode derivedClass)
 
       _derivedClasses.Add(derivedClass);
     }
+
+    private bool ImplementsInterface (InterfaceDefinitionBuilderNode interfaceDefinitionBuilderNode)
+    {
+      if (ImplementedInterfaces.Contains(interfaceDefinitionBuilderNode))
+        return true;
+
+      if (IsInheritanceRoot)
+        return false;
```

Comment:
I think we can drop this. BaseClass already covers this case and removes all the questions about whether interitanceroot has interfaces or not (i.e. it's not relevant here?)
```suggestion
```

Reply:
I don't think that we should drop this check here. If a node has `IsInteritanceRoot == true` then its `TypeDefinition` is not linked to the parent as is the case with the node. As such, the check in the node also should not ask the parent if it implements a certain interface. For our current scenario it might be okay to drop, but I think that the current check better represents the behavior we want.

Reply:
I would like to discuss this live. Those rules are a bit hard to wrap the mind around.